pkg/addon/controllers/addonconfiguration/graph.go

```diff
@@ -48,14 +48,14 @@ type addonNode struct {
 	status         *clustersdkv1alpha1.ClusterRolloutStatus
 }
 
-type addonConfigMap map[addonv1alpha1.ConfigGroupResource]addonv1alpha1.ConfigReference
+type addonConfigMap map[addonv1alpha1.ConfigGroupResource][]addonv1alpha1.ConfigReference
```

Comment:
would it be easier if we use something like 
`type addonConfigMap map[addonv1alpha1.ConfigGroupResource]set.Set[addonv1alpha1.ConfigReference]`?

Reply:
For the ConfigReference, we use it to store ConfigGroupResource, DesiredConfig, LastObservedGeneration and LastAppliedConfig, while only use ConfigGroupResource and DesiredConfig.ConfigReferent when doing comparison, it seems not be able to use the strengths of sets, like Has().